walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/LinkedDomainsService.kt

```diff
@@ -7,47 +7,53 @@ import com.microsoft.walletlibrary.did.sdk.credential.service.models.linkedDomai
 import com.microsoft.walletlibrary.did.sdk.identifier.resolvers.Resolver
 import com.microsoft.walletlibrary.did.sdk.util.Constants
-import com.microsoft.walletlibrary.did.sdk.util.controlflow.Result
-import com.microsoft.walletlibrary.did.sdk.util.controlflow.map
-import com.microsoft.walletlibrary.did.sdk.util.controlflow.runResultTry
+import com.microsoft.walletlibrary.did.sdk.util.controlflow.SdkException
 import com.microsoft.walletlibrary.did.sdk.util.log.SdkLog
 import java.net.URL
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 internal class LinkedDomainsService @Inject constructor(
-    private val apiProvider: ApiProvider,
+    private val apiProvider: HttpAgentApiProvider,
     private val resolver: Resolver,
     private val jwtDomainLinkageCredentialValidator: DomainLinkageCredentialValidator
 ) {
     suspend fun fetchAndVerifyLinkedDomains(relyingPartyDid: String): Result<LinkedDomainResult> {
-        return runResultTry {
-            val domainUrls = getLinkedDomainsFromDid(relyingPartyDid).abortOnError()
-            verifyLinkedDomains(domainUrls, relyingPartyDid)
-        }
+        getLinkedDomainsFromDid(relyingPartyDid)
+            .onSuccess { domainUrls ->
+                return verifyLinkedDomains(domainUrls, relyingPartyDid)
+            }
+            .onFailure {
+                return Result.failure(it)
+            }
+        return Result.failure(SdkException(""))
     }
 
     private suspend fun verifyLinkedDomains(domainUrls: List<String>, relyingPartyDid: String): Result<LinkedDomainResult> {
-        return runResultTry {
-            if (domainUrls.isEmpty())
-                return@runResultTry Result.Success(LinkedDomainMissing)
-            val domainUrl = domainUrls.first()
-            val hostname = URL(domainUrl).host
-            val wellKnownConfigDocumentResult = getWellKnownConfigDocument(domainUrl)
-            if (wellKnownConfigDocumentResult is Result.Success) {
-                val wellKnownConfigDocument = wellKnownConfigDocumentResult.payload
-                wellKnownConfigDocument.linkedDids.forEach { linkedDidJwt ->
+        if (domainUrls.isEmpty())
+            return Result.success(LinkedDomainMissing)
+        val domainUrl = domainUrls.first()
+        val hostname = URL(domainUrl).host
+        return getWellKnownConfigDocument(domainUrl)
+            .map {
+                wellKnownConfigDocument ->
+                wellKnownConfigDocument.linkedDids.firstNotNullOf { linkedDidJwt ->
                     val isDomainLinked = jwtDomainLinkageCredentialValidator.validate(linkedDidJwt, relyingPartyDid, domainUrl)
-                    if (isDomainLinked) return@runResultTry Result.Success(LinkedDomainVerified(hostname))
+                    if (isDomainLinked)
+                        LinkedDomainVerified(hostname)
+                    else
+                        null
```

Comment:
If the linked domain being processed is not verified, shouldn't next one be processed instead of returning null?